include/linux/util_macros.h

```diff
@@ -45,35 +29,41 @@
  * order. The iteration is done in reverse order, so that the comparison
  * of '__fc_right' & '__fc_left' also works for unsigned numbers.
  */
-#define find_closest_descending(x, a, as)				\
-({									\
-	typeof(as) __fc_i, __fc_as = (as) - 1;				\
-	long __fc_mid_x, __fc_x = (x);					\
-	long __fc_left, __fc_right;					\
-	typeof(*a) const *__fc_a = (a);					\
-	for (__fc_i = __fc_as; __fc_i >= 1; __fc_i--) {			\
-		__fc_mid_x = (__fc_a[__fc_i] + __fc_a[__fc_i - 1]) / 2;	\
-		if (__fc_x <= __fc_mid_x) {				\
-			__fc_left = __fc_x - __fc_a[__fc_i];		\
-			__fc_right = __fc_a[__fc_i - 1] - __fc_x;	\
-			if (__fc_right < __fc_left)			\
-				__fc_i--;				\
-			break;						\
-		}							\
-	}								\
-	(__fc_i);							\
-})
+#define find_closest_descending(x, a, as) \
+	find_closest_descending_s32((s32)(x), (const s32 *)(a), (as))
 
 /**
  * is_insidevar - check if the @ptr points inside the @var memory range.
- * @ptr:	the pointer to a memory address.
- * @var:	the variable which address and size identify the memory range.
  *
- * Evaluates to true if the address in @ptr lies within the memory
- * range allocated to @var.
+ * @ptr:  the pointer to be checked.
+ * @var:  the variable which memory range is checked against.
+ * @size: the size of variable, if unspecified it will be determined by sizeof().
+ *
+ * Return 1 if the @ptr lies within the @var memory range, 0 if outside or
+ * on the boundary.
  */
-#define is_insidevar(ptr, var)						\
-	((uintptr_t)(ptr) >= (uintptr_t)(var) &&			\
-	 (uintptr_t)(ptr) <  (uintptr_t)(var) + sizeof(var))
+#define is_insidevar(ptr, var, ...)					\
```

Comment:
this is also patch-noise
remove it;

if wanting to fix/address this, then it should be a separate patch;